src/webgpu/api/validation/createView.spec.ts

```diff
@@ -101,19 +97,48 @@ g.test('creating_texture_view_on_a_2D_non_array_texture')
     // mip level is in range
     { mipLevelCount: 1, baseMipLevel: MIP_LEVEL_COUNT - 1, _success: true },
     { mipLevelCount: 2, baseMipLevel: MIP_LEVEL_COUNT - 2, _success: true },
-    // baseMipLevel == k && mipLevelCount == 0 means to use levels k..end
-    { mipLevelCount: 0, baseMipLevel: 0, _success: true },
-    { mipLevelCount: 0, baseMipLevel: 1, _success: true },
-    { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT - 1, _success: true },
-    { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT, _success: false },
+    // mipLevelCount == 0 is invalid
+    { mipLevelCount: 0, baseMipLevel: 0, _success: false },
+    // baseMipLevel == k && mipLevelCount == undefined means to use levels k..end
+    {
+      mipLevelCount: undefined,
+      baseMipLevel: 0,
+      _mipLevelCountKeepUndefined: true,
```

Comment:
I don't know Typescript so well, but can't we just skip this variable and have mipLevelCount: undefined mean that?

Reply:
Hmm, the helper function above explicitly assigns values to mipLevelCount and arrayLayerCount.

Reply:
Which helper function? `params`?

Reply:
GetDescriptor() at line 59.

Reply:
Ah, got it thanks! Either way sounds find. Maybe @kainino0x you have a suggestion to make this more idiomatic?

Reply:
Thanks for your detailed suggestion, Kai. I looked further for the current validation tests for createTexture and creatView. And I totally agree with you that the tests are very old and we need to add a lot more tests and we may need significant changes to these two tests. I will revisit the tests and revise them incrementally (at the same time, I will work on 3D texture feature implementation in Dawn project, like copy from/to 3D texture, etc.), not just add a couple of random tests for 3D texture validation. Let's discuss the issues you just mentioned in the coming PRs. 

And I'd like to close this PR.

Reply:
I don't think it's necessary to redo everything to incrementally fix these outdated cases, but on the other hand maybe we want to redesign the testing approach anyway. Either way is OK with me - if we redesign, we should consider both validation and operation testing at once, have a test plan review, and try to make sure it's covering all the same stuff.
We could also design new tests but keep these old tests; test duplication doesn't really hurt. If we want to keep these then we'll have to do incremental updates to make them correct (like the suggested code block above).

Reply:
Thanks for the suggestion. I looked into the existing test plan for createTexture, it is good. I didn't find significant missing parts. But the tests have not been added yet. The tests lag far behind.